be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -117,9 +119,24 @@ Status JsonScanner::_construct_json_types() {
             child_type->children.emplace_back(varchar_type);
 
             _json_types[column_pos] = std::move(json_type);
-        } else {
+            break;
+        }
+
+        // Treat BIGINT(8B), INT(4B), SMALLINT(2B), TINYINT(1B) as BIGINT(8B).
+        case TYPE_BIGINT:
+        case TYPE_INT:
+        case TYPE_SMALLINT:
+        case TYPE_TINYINT: {
+            _json_types[column_pos] = TypeDescriptor{slot_desc->type().type};
```

Comment:
The code doesn't look consistent with the comments.
Comment says treat them as BIGINT, however code treat them as what they are.

Reply:
Corrected.